e2e/scenario/validators.go

```diff
@@ -103,6 +103,72 @@ func NonEmptyDirectoryValidator(dirName string) *LiveVMValidator {
 	}
 }
 
+func FileHasContentsValidator(fileName string, contents string) *LiveVMValidator {
+	return &LiveVMValidator{
+		Description: fmt.Sprintf("Assert that %s has defined contents", fileName),
+		Command:     fmt.Sprintf("ls -la %[1]s && cat %[1]s | grep -q '%[2]s'", fileName, contents),
+		Asserter: func(code, stdout, stderr string) error {
+			if code != "0" {
+				return fmt.Errorf("expected to find a file '%s' with contents '%s' but did not", fileName, contents)
+			}
+			return nil
+		},
+	}
+}
+
+// this function is just used to remove some bash specific tokens so we can echo the command to stdout.
+func cleanse(str string) string {
+	str = strings.Replace(str, "'", "", -1)
+	str = strings.Replace(str, "\"", "", -1)
+	str = strings.Replace(str, "(", "OB", -1)
+	str = strings.Replace(str, ")", "CB", -1)
+	str = strings.Replace(str, "||", "OR", -1)
+	str = strings.Replace(str, "&&", "AND", -1)
+
+	return str
+}
+
+func serviceCanRestartValidator(serviceName string, restartTimeoutInSeconds int) *LiveVMValidator {
+	steps := []string{
```

Comment:
maybe we should investigate whether we can just use dedicated scripts to accomplish this at some point

Reply:
Oh absolutely. I'd love to have this specified in a yaml file so we can add new ones easily. This code is a bit of a hack TBH.